libs/designer/src/lib/core/state/undoRedo/undoRedoSlice.ts

```diff
@@ -1,33 +1,57 @@
 import { createSlice, type PayloadAction } from '@reduxjs/toolkit';
 import { resetWorkflowState } from '../global';
-import type { StateHistory } from './undoRedoTypes';
+import type { StateHistory, StateHistoryItem } from './undoRedoTypes';
 
 export const initialState: StateHistory = {
   past: [],
   future: [],
+  stateHistoryItemIndex: -1,
 };
 
 export const undoRedoSlice = createSlice({
   name: 'undoRedo',
   initialState,
   reducers: {
-    saveStateToHistory: (state, action: PayloadAction<{ compressedState: Uint8Array; limit: number }>) => {
-      if (action.payload.limit < 1) {
-        return;
-      }
+    saveStateToHistory: (state, action: PayloadAction<{ stateHistoryItem: StateHistoryItem; limit: number }>) => {
       state.past = [
         ...(state.past.length < action.payload.limit ? state.past : state.past.slice(1, state.past.length)),
-        action.payload.compressedState,
+        { ...action.payload.stateHistoryItem },
       ];
       state.future = [];
+      state.currentEditedPanelTab = undefined;
+      state.currentEditedPanelNode = undefined;
     },
-    updateStateHistoryOnUndoClick: (state, action: PayloadAction<Uint8Array>) => {
+    updateStateHistoryOnUndoClick: (state, action: PayloadAction<StateHistoryItem>) => {
+      // Add current edited panel details to state being added to history
+      const newStateItem: StateHistoryItem = {
+        compressedState: action.payload.compressedState,
+        editedPanelTab: state.currentEditedPanelTab,
+        editedPanelNode: state.currentEditedPanelNode,
+      };
+
+      // Update current edited panel details to the one from the state we are undoing to
+      // This ensures we don't lose track of edited panel details on undo/redo click when state is updated
+      state.currentEditedPanelTab = state.past[state.past.length - 1].editedPanelTab;
```

Comment:
Shouldn't this `editedPanelTab` already be in the panel state?  I'm confused on why we need to do this manually, if we load a compressed state the selected panel tab + node should be loaded as well

Reply:
Discussed offline